core/store/src/cold_storage.rs

```diff
@@ -22,6 +24,28 @@ struct StoreWithCache<'a> {
     cache: StoreCache,
 }
 
+/// Asynchronously writes set operations to `cold_db` in batches.
+/// Collects operation in batch, until its size is bigger than `min_transaction_size`.
+/// Then creates a thread to write that information to `cold_db`.
+/// Then joins some old thread,
+/// until size of transactions in `handles` is smaller than `max_handles_size`.
+/// Takes care of writing information to `cold_db`,
+/// [`write`] and [`join`] for all handles will be called on destruction.
+struct BatchTransaction<D: Database + 'static> {
+    cold_db: std::sync::Arc<ColdDB<D>>,
+    transaction: DBTransaction,
+    /// Size of all values keys and values in `transaction` in bytes.
+    transaction_size: usize,
+    /// Size of all values keys and values in `handles` in bytes.
+    handles_size: usize,
+    /// Maximum size, that we keep by joining front elements of handles.
+    max_handles_size: usize,
+    /// Minimum size, after which we write transaction
+    min_transaction_size: usize,
+    /// Currently not joined write threads.
+    handles: VecDeque<(JoinHandle<io::Result<()>>, usize)>,
```

Comment:
We may also want to limit the number of threads that we spawn, especially when min_transaction_size is 0.

Reply:
Will remove threads logic from this PR for simplicity